client/me/purchases/manage-purchase/index.jsx

```diff
@@ -745,7 +753,9 @@ export default connect( ( state, props ) => {
 	return {
 		hasLoadedDomains,
 		hasLoadedSites,
-		hasLoadedUserPurchasesFromServer: hasLoadedUserPurchasesFromServer( state ),
+		hasLoadedUserPurchasesFromServer: selectedSiteId
+			? hasLoadedSitePurchasesFromServer( state )
+			: hasLoadedUserPurchasesFromServer( state ), // TODO: refactor this
```

Comment:
About this todo: this prop is eventually passed with this name to a utility component. It would be nice to rename the prop.

Reply:
Nice find. From what I can tell, this prop is passed to `RemovePurchase` which then passes it to `isDataLoading` (from `client/me/purchases/utils.js`) which then uses it to determine if it should return true or false. `isDataLoading` is used in several other places within `/me/purchases`. `RemovePurchase` is also used in one other place (`client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx`).

My suggestion, and it could be done in this PR, would be to change this prop to `hasLoadedPurchases`, then pass that prop as `hasLoadedUserPurchasesFromServer={ this.props.hasLoadedPurchases }` to `RemovePurchase`. 

That is, the proper solution is probably to modify `RemovePurchase` and `isDataLoading` to also take into account the site purchases boolean, but that would mean quite a few things to change. Instead, since it looks safe to pretend that the variable is something else (as you've already done here), we continue to do that, but make it explicit within this component rather than hiding it here in `mapStateToProps` where it would be easier to miss if someone was examining this component in the future. Basically, this is a (small) hack and it's good to make those obvious.